validator/client/log.go

```diff
@@ -45,16 +45,23 @@ func (v *validator) LogNextDutyTimeLeft(slot uint64) error {
 	}
 
 	var nextDutySlot uint64
-	var role string
+	attestingCount := 0
+	proposingCount := 0
 	for _, duty := range v.duties.CurrentEpochDuties {
-		if duty.AttesterSlot > slot && (nextDutySlot > duty.AttesterSlot || nextDutySlot == 0) {
+		if duty.AttesterSlot == nextDutySlot {
+			attestingCount += 1
+		} else if duty.AttesterSlot > slot && (nextDutySlot > duty.AttesterSlot || nextDutySlot == 0) {
 			nextDutySlot = duty.AttesterSlot
-			role = "attester"
+			attestingCount = 1
+			proposingCount = 0
```

Comment:
```suggestion
```

Reply:
The original logic here is to loop through all `AttesterSlot` and `ProposerSlots` of the duties, keep updating `nextDutySlot` if we find a smaller future `slot` that has `attesting`/`proposing` duty. So to extend this idea to count number of `attesting`/`proposing` within the same loop, whenever `nextDutySlot` gets changed, the `attestingCount` and `proposingCount` needs to be reset (as those counters are counting for the old `nextDutySlot`). The reset is base on what duty causing `nextDutySlot` to update, if it is because of `duty.AttesterSlot < nextDutySlot`, we reset counters to  `attestingCount = 1; proposingCount = 0`, if it is because of `proposerSlot < nextDutySlot`, we reset counters to `attestingCount = 0; proposingCount = 1`.

Reply:
What happens if we get proposer and attesters on the same slot? Looks like the previous design has the same problem as well

Manipulating `attestingCount` and `proposingCount` are harder to reason. I think it might be better if we use a map `dutyCount` that is `dutySlot -> [attestingCount, proposingCount]`

```go
logrus.Fields{
					"attesting":   dutyCount[dutySlot][0],
					"proposing":   dutyCount[dutySlot][1],
				}).Info("Next duty")
		}
```

What do you think?

Reply:
Previous design can show either `role=attester` or `role=proposer`, depends on which appears first in the `duties` array, if we get proposer and attesters on the same slot.

Your approach is easier to understand, and in fact, it aligns with the `Alternative Solution` described in #8118. Instead of just reporting the duties of `nextDutySlot`, it can be more useful for maintenance planning if we display `dutyCount` of all slots in current epoch. Your `dutyCount` map extends nicely to that.

I am fine to do the `dutyCount` map though it contains extra info the `Next duty` log doesn't need at the moment.

Reply:
@terencechain I have made the changes, it is much easier to understand, thanks for your advice!